glean-core/android/src/main/java/mozilla/telemetry/glean/Glean.kt

```diff
@@ -197,19 +197,21 @@ open class GleanInternalAPI internal constructor () {
                 initializeCoreMetrics(applicationContext)
             }
 
-            // Signal Dispatcher that init is complete
-            Dispatchers.API.flushQueuedInitialTasks()
-
             // Check if the "dirty flag" is set. That means the product was probably
             // force-closed. If that's the case, submit a 'baseline' ping with the
             // reason "dirty_startup". We only do that from the second run.
+            // This must be done before flushing the queued initial tasks so that
+            // it is always sent before any baseline ping with reason "foreground".
             if (!isFirstRun && LibGleanFFI.INSTANCE.glean_is_dirty_flag_set().toBoolean()) {
                 submitPingByNameSync("baseline", "dirty_startup")
                 // Note: while in theory we should set the "dirty flag" to true
                 // here, in practice it's not needed: if it hits this branch, it
                 // means the value was `true` and nothing needs to be done.
             }
 
+            // Signal Dispatcher that init is complete
+            Dispatchers.API.flushQueuedInitialTasks()
```

Comment:
This is basically matching what you're fixing with #810 , rigth?

Reply:
No, slightly different.  We need to send the `dirty_startup` ping before the `foreground` ping, so that any ping-lifetime metrics on the baseline ping are sent correctly in the `dirty_startup` ping.